### PR TITLE
Adds data source properties to elasticsearch, wikipedia and google-drive

### PR DESCRIPTION
**Pull request by rvztz**
Addresses #1316.

Adds data source properties to elasticsearch, wikipedia and google-drive:
-  data_created
-  date_modified
-  version
-  record_locator
These properties are instantiated when supported by the connector.

Separates the logic between fetching the file from source and `get_file`. Retrieves file metadata when any of the properties are called.

Adds logic to check if file exists in the remote source. For connectors that don't directly support it, adds exception handling to check any issues while retrieving the file.

### CODE REVIEW

File: unstructured/ingest/connector/elasticsearch.py [hunk 1]
```diff
@@ -104,21 +105,52 @@ def _concatenate_dict_fields(self, dictionary, seperator="\n"):
...
+            es = Elasticsearch(self.connector_config.url)
+            document = es.get(
+                index=self.connector_config.index_name,
+                id=self.document_meta.document_id,
+            )
+        except NotFoundError:
+            logger.error("Couldn't find document with ID: %s", self.document_meta.document_id)
+            return None
+        except Exception:
+            raise
```
Comment (ryannikolaidis) [inline unstructured/ingest/connector/elasticsearch.py:124]: why this? if we just don't catch the exception here, this should happen automatically right?

**Review by ryannikolaidis — APPROVED** [review]
lgtm!

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,7 @@
 * **Better debug output related to sentence counting internals**. Clarify message when sentence is not counted toward sentence count because there aren't enough words, relevant for developers focused on `unstructured`s NLP internals.
 * **Faster ocr_only speed for partitioning PDF and images.** Use `unstructured_pytesseract.run_and_get_multiple_output` function to reduce the number of calls to `tesseract` by half when partitioning pdf or image with `tesseract`
 * **Adds data source properties to fsspec connectors** These properties (date_created, date_modified, version, source_url, record_locator) are written to element metadata during ingest, mapping elements to information about the document source from which they derive.
+* **Adds data source properties (date_created, date_modified, version, exists, source_url, record_locator) to the Elasticsearch, Google Drive and Wikipedia connectors.** Implements `update_source_metadata` method for the Elasticsearch, Google Drive and Wikipedia connectors connector.
```
Comment (ryannikolaidis) [inline CHANGELOG.md:15]:
```suggestion
* **Adds data source properties to connectors** These properties (date_created, date_modified, version, source_url, record_locator) are written to element metadata during ingest, mapping elements to information about the document source from which they derive.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,6 +12,7 @@
 * **Better debug output related to sentence counting internals**. Clarify message when sentence is not counted toward sentence count because there aren't enough words, relevant for developers focused on `unstructured`s NLP internals.
 * **Faster ocr_only speed for partitioning PDF and images.** Use `unstructured_pytesseract.run_and_get_multiple_output` function to reduce the number of calls to `tesseract` by half when partitioning pdf or image with `tesseract`
 * **Adds data source properties to fsspec connectors** These properties (date_created, date_modified, version, source_url, record_locator) are written to element metadata during ingest, mapping elements to information about the document source from which they derive.
+* **Adds data source properties to connectors** These properties (date_created, date_modified, version, source_url, record_locator) are written to element metadata during ingest, mapping elements to information about the document source from which they derive.
```
Comment (cragwolfe) [inline CHANGELOG.md:15]:
```suggestion
* **Adds data source properties to connectors** These properties (date_created, date_modified, version, source_url, record_locator) are written to element metadata during ingest, mapping elements to information about the document source from which they derive. The source_url may be especially useful for downstream applicaton for a link back to the original document, e.g. a GDrive doc or Notion page.
```

**Pull request merged**